### DISCUSSION TITLE
How to get common web frameworks running?

### DISCUSSION THREAD

**Discussion by janus-reith** [wasmCloud/wasmCloud#1786] [Q&A]
With the recent release of [WinterJS](https://github.com/wasmerio/winterjs) by Wasmer and the competing standards that seem to exist in the WASM ecosystem I was wondering how complex the wiring would be to get that up and running on wasmcloud, or perhaps if it might even work out of the box if compiled in the proper format instead of Wasmer, or if it does not fit the way wasmcloud functions are built at all - or if something similar to what WinterJS does exists in a wasmcloud compatible shape already?

My primary use case would be to deploy a limited in scope distributed FaaS environment similar to Cloudflare Workers where I could run Nextjs and Astro apps, without relying on Kubernetes.

**Comment by brooksmtownsend** [comment]
@janus-reith this is a really interesting question, when it comes to executing functions we do support that well today with a combination of the WebAssembly component model standard and the https://github.com/bytecodealliance/ComponentizeJS project. We do have a starting tutorial with TypeScript, but that works just the same with JavaScript! https://wasmcloud.com/docs/tour/hello-world?lang=typescript#creating-a-project

Now for your second comment, around your primary use case, we haven't explored compiling Nextjs/Astro apps to Wasm components, but that would be an extremely fun experiment. @lachieh do you have any thoughts here on what may be possible here? Server-side logic running in the component sounds perfect.

(as a side note, thank you for your patience @janus-reith, I didn't have notifications turned on for discussions!)

**Comment by brooksmtownsend** [comment]
Hey @janus-reith , you may be interested in https://wasmcloud.com/blog/2024-12-10-leptos-and-wasi-0.2-for-full-stack-wasm-development 😄 

I'm going to close this issue since it hasn't been active in a bit, but please feel free to leave comments / reopen for more discussion
